File: aula4/ContaBancaria.py
class ContaBancaria:
    def __init__(self, nome, saldo_inicial):
        self.titular = nome
        self.saldo = saldo_inicial

    def exibir_saldo(self):
        print(f"Titular: {self.titular}, Saldo: {self.saldo}")

    def sacar(self, valor):
        if valor <= self.saldo:
            self.saldo -= valor
            print(f"Saque de {valor} BRL realizado com sucesso.")
        else:
            print(f"Saldo insuficiente para o saque de {valor} BRL.")

    def sacar_com_taxa_cambio(self, valor, taxa_cambio):
        valor_convertido = valor * taxa_cambio
        if valor_convertido <= self.saldo:
            self.saldo -= valor_convertido
            print(f"Saque de {valor_convertido} BRL realizado com sucesso.")
        else:
            print(f"Saldo insuficiente para o saque de {valor_convertido} BRL.")

    def sacar_com_tipo_operacao(self, valor, tipo_operacao):
        taxa = 0.05 if tipo_operacao == "credito" else 0.0
        valor_com_taxa = valor + (valor * taxa)
        if valor_com_taxa <= self.saldo:
            self.saldo -= valor_com_taxa
            print(f"Saque de {valor} BRL realizado com taxa de {valor * taxa} BRL.")
        else:
            print("Saldo insuficiente para o saque.")

# Teste da classe
if __name__ == "__main__":
    conta = ContaBancaria("Josevaldo", 5000.00)

    conta.exibir_saldo()
    conta.sacar(1000)
    conta.exibir_saldo()
    conta.sacar_com_taxa_cambio(1000, 5.4)
    conta.exibir_saldo()
    conta.sacar_com_tipo_operacao(1000, "credito")
    conta.exibir_saldo()

File: 1semana/aula4/ContaBancaria.java
class ContaBancaria {
    private String titular;
    private double saldo;

    public ContaBancaria(String nome, double saldoInicial) {
        this.titular = nome;
        this.saldo = saldoInicial;
    }

    public void exibirSaldo() {
        System.out.println("Titular: " + titular + ", Saldo: " + saldo);
    }

    public void sacar(double valor) {
        if (valor <= saldo) {
            saldo -= valor;
            System.out.println("Saque de " + valor + " BRL realizado com sucesso.");
        } else {
            System.out.println("Saldo insuficiente para o saque de " + valor + " BRL.");
        }
    }

    public void sacar(double valor, double taxaCambio) {
        double valorConvertido = valor * taxaCambio;
        if (valorConvertido <= saldo) {
            saldo -= valorConvertido;
            System.out.println("Saque de " + valorConvertido + " BRL realizado com sucesso.");
        } else {
            System.out.println("Saldo insuficiente para o saque de " + valorConvertido + " BRL.");
        }
    }

    public void sacar(double valor, String tipoOperacao) {
        double taxa = tipoOperacao.equals("credito") ? 0.05 : 0.0;
        double valorComTaxa = valor + (valor * taxa);
        if (valorComTaxa <= saldo) {
            saldo -= valorComTaxa;
            System.out.println("Saque de " + valor + " BRL realizado com taxa de " + (valor * taxa) + " BRL.");
        } else {
            System.out.println("Saldo insuficiente para o saque.");
        }
    }

    public static void main(String[] args) {
        ContaBancaria conta = new ContaBancaria("Josevaldo", 5000.00);

        conta.exibirSaldo();
        conta.sacar(1000);
        conta.exibirSaldo();
        conta.sacar(1000, 5.4);
        conta.exibirSaldo();
        conta.sacar(1000, "credito");
        conta.exibirSaldo();
    }
}